src/czbenchmarks/tasks/base.py

```diff
@@ -111,70 +57,39 @@ def _compute_metrics(self) -> List[MetricResult]:
-                  output data
-            model_types: Optional list of specific model types to evaluate. If None,
-                         will use all available model types common across datasets.
-
+            cell_representation: gene expression data or embedding for task
+            task_kwargs: Additional arguments passed to the task
+            metric_kwargs: Additional arguments passed to the metrics
         Returns:
-            Dictionary mapping model types to their list of MetricResult objects
+            List of MetricResult objects
 
-        Raises:
-            ValueError: If no common model types found across datasets or
-                       if a specified model type is not available in a dataset
         """
-        # Dictionary to store metrics for each model type
-        all_metrics_per_model = {}
-
-        # Determine which model types to evaluate if not explicitly provided
-        if model_types is None:
-            if isinstance(data, list):
-                # For multiple datasets, find model types available in all datasets
-                model_types = set.intersection(
-                    *[set(dataset.outputs.keys()) for dataset in data]
-                )
-                if not model_types:
-                    raise ValueError("No common model types found across all datasets")
-            else:
-                # For single dataset, use all available model types
-                model_types = list(data.outputs.keys())
-
-        # Validate that all requested model types are available in all datasets
-        for model_type in model_types:
-            if isinstance(data, list):
-                for dataset in data:
-                    if model_type not in dataset.outputs:
-                        raise ValueError(
-                            f"Model type {model_type} not found in dataset"
-                        )
-            else:
-                if model_type not in data.outputs:
-                    raise ValueError(f"Model type {model_type} not found in dataset")
-
-        # Process each model type
-        for model_type in model_types:
-            # Run the task implementation for this model
-            self._run_task(data, model_type)
-
-            # Compute metrics based on task results
-            metrics = self._compute_metrics()
-
-            # Store metrics for this model type
-            all_metrics_per_model[model_type] = metrics
-
-        return all_metrics_per_model
-
-    def set_baseline(self, data: BaseDataset, **kwargs):
+
+        task_output = self._run_task(cell_representation, **task_kwargs)
+
+        # Handle cases where embedding required by metrics but not set by _run_task
+        if "cell_representation" not in metric_kwargs:
+            task_output.setdefault("cell_representation", cell_representation)
+
+        metrics = self._compute_metrics(**task_output, **metric_kwargs)
+        return metrics
+
+    def set_baseline(
```

Comment:
This isn't setting anything. It's computing. So maybe rename to something like `compute_baseline`